src/lib/storage/base_run_length_segment.hpp

```diff
@@ -0,0 +1,22 @@
+#pragma once

```

Comment:
Not a big fan of that class. Is it only necessary because we want to import/export binary run length segments?

Reply:
Indeed the class is only necessary for exporting, as the concretion of the segment encoding is done via a visitor pattern. This would likely be omitted if every segment would encode itself, which on the other hand would probably mean a deeper design decision.

Reply:
I think we should get rid of this class and all the visitors altogether. The visitor pattern has been removed from pretty much all other parts of Hyrise and I'd vote to remove it here as well.

I think a simple method that resolves the encoding type and forwards the segment accordingly (see `ChunkEncoder::encode_segment()`) should suffice.